tap_salesforce/__init__.py

```diff
@@ -51,19 +51,21 @@ def main_impl():
     stream = Stream(args.state)
     sync_fields(sf, stream)
 
-    for stream_id, fields, replication_key in sf.get_tables():
+    for table, fields, replication_key in sf.get_tables():
         if not fields:
             LOGGER.info(
-                f"skipping stream {stream_id} since it does not exist on this account"
+                f"skipping stream {table.name} since it does not exist on this account"
             )
             continue
 
-        LOGGER.info(f"processing stream {stream_id}")
+        LOGGER.info(f"processing stream {table.name}")
 
-        start_time = stream.get_stream_state(stream_id, replication_key) or config_start
+        start_time = (
+            stream.get_stream_state(table.name, replication_key) or config_start
```

Comment:
Just double checking: do we actually use table names for state?

Reply:
yes. Here is an example 
```
{
  "bookmarks": {
    "Account": { "LastModifiedDate": "2023-01-25T01:56:41Z" },
    "AccountContactRelation": { "LastModifiedDate": "2023-01-25T00:27:33Z" },
    "AccountHistory": { "CreatedDate": "2023-01-25T01:52:37Z" },
    "Campaign": { "LastModifiedDate": "2023-01-24T23:22:55Z" },
    "CampaignMember": { "LastModifiedDate": "2023-01-25T01:05:40Z" },
    "Contact": { "LastModifiedDate": "2023-01-25T02:01:47Z" },
    "ContactHistory": { "CreatedDate": "2023-01-25T02:01:47Z" },
    "Event": { "LastModifiedDate": "2023-01-24T21:03:52Z" },
    "Lead": { "LastModifiedDate": "2023-01-24T23:45:00Z" },
    "LeadHistory": { "CreatedDate": "2023-01-24T20:30:50Z" },
    "Opportunity": { "LastModifiedDate": "2023-01-25T02:00:11Z" },
    "OpportunityContactRole": { "LastModifiedDate": "2023-01-25T01:08:47Z" },
    "OpportunityFieldHistory": { "CreatedDate": "2023-01-25T01:08:47Z" },
    "OpportunityHistory": { "CreatedDate": "2023-01-25T01:08:49Z" },
    "OpportunityLineItem": { "LastModifiedDate": "2023-01-25T01:08:47Z" },
    "Product2": { "LastModifiedDate": "2022-12-08T13:12:42Z" },
    "RecordType": { "LastModifiedDate": "2023-01-24T21:10:23Z" },
    "Task": { "LastModifiedDate": "2023-01-25T02:04:23Z" },
    "User": { "LastModifiedDate": "2023-01-24T15:51:55Z" },
    "UserRole": { "LastModifiedDate": "2023-01-20T17:57:08Z" }
  }
}
```